### PR TITLE
Add introduction to C++

### PR DESCRIPTION
**Pull request by ElspethBair**

### CODE REVIEW

File: robot/rookie/c++/README.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+example:
+    
+    double age = 15;
+
+We have to tell the computer that the variable age is a double, because is won't recognize that itself. 
+
+**Console Input/Output**
+
+Including <iostream> allows you to use std::cin and std::cout. std::cout is the C++ equivalent of the print() function in Python. std::cin is the C++ equivalent of the input() function.
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: use \`code\`

File: robot/rookie/c++/README.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
+#Introduction to C++
+Here you will learn the basics of the main language that robot programming uses, C++
+
+###C++ Program Structure
+
+Every C++ program starts with *include statements*. They let you use functions and classes that have been declared in other *header files*.
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: mention that these are the equivalent of `import`.

File: robot/rookie/c++/README.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+Every C++ program starts with *include statements*. They let you use functions and classes that have been declared in other *header files*. 
+
+  example:
+    
+    #include <iostream>
+
+(The header <iostream> gives you access to input and output helpers, like std::cin and std::out. )
+
+Another important part of your program is the *main function definition*. This defines a function called "main" that tells the computer where to begin executing code.
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: It's important to emphasize that `int main()` is a special function

File: robot/rookie/c++/README.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+Another important part of your program is the *main function definition*. This defines a function called "main" that tells the computer where to begin executing code. 
+
+At the end of, but still inside your main function, add a return statement:
+    
+    return 0;
+
+At this point, the return statement is not technically required, but it is good practice. This is an error code for your program - a value of 0 tells the computer that the program has finished executing successfully. 
+
+The next, very important part of your program is the *executable code*. This is the code that actually runs in your program. The executable code is inside your main function.
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: have an actual example of a simple program here

File: robot/rookie/c++/README.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+      def greet(self,name):
+         return "Hi, “,name,”! I'm ”,self.firstName, 
+           self.lastName,”!"
+
+C++:
+
+     class Person {
+      public:
+       // Initialize first_name and last_name
+       Person(std::string first, std::string last) :
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: It's important that the constructor has the same name as the class

File: robot/rookie/c++/README.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+      public:
+       // Initialize first_name and last_name
+       Person(std::string first, std::string last) : 
+       first_name(first), last_name(last) {}
+
+       std::string Greet(std::string name) {
+          return "Hi, " + name + "! I'm " + first_name + " " + last_name + "!";
+    }
+
+     private:
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: indent line 103 with 2 more spaces and line 106 with one less.

File: robot/rookie/c++/README.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+       std::string first_name, last_name;
+    };
+
+
+**Dynamic-size Arrays**
+
+A dynamic size array is an array that can be added to or taken away from. They are used to store values.
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: b sure you emphasize the difference between vectors and arrays. Also, in the general python training, they should be learning about lists, not arrays.

File: robot/rookie/c++/README.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    // Iteration
+    for n in nums:
+      print(n)
+
+    // Check size
+    println(len(num))
+
+    nums.append(7) // Add value
+
+(std::vector is defined in <vector> so remember to add it in your includes!)
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: use \`\` for code

File: robot/rookie/c++/README.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+
+     print (squared(5))
+
+ the same function in C++:
+
+     int Squared(int x) {
+     	return x * x;
+     }
+
+     std::cout << Squared(5) << std::endl;
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: https://stackoverflow.com/questions/213907/c-stdendl-vs-n I'm pretty sure it doesn't matter which one you use, but be sure to explain what `std::endl` is before using it.

File: robot/rookie/c++/README.md [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+    };
+
+
+**Dynamic-size Arrays**
+
+A dynamic size array is an array that can be added to or taken away from. They are used to store values. 
+C++:
+
+    auto nums = std::vector<int> 
+    {1,2,3,5,7,11};
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: that should fit on one line

File: robot/rookie/c++/README.md [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+    std::vector<std::string> strings =	
+      {"hello", "how are you", "goodbye"};
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: same here

File: robot/rookie/c++/README.md [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+
+C++:
+
+    std::string msg = "A string in C++";
+
+Python: 
+
+    msg = "A string in Python"
+
+Strings are in the std namespace like cout and cin, so when you want to use one, you have to type std::string. std::string is defined in <string>, so you have to put it in your includes at the beginning of your program if are using strings.
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: I think the standard library is in the next lesson, and we haven't covered namespaces yet.

File: robot/rookie/c++/README.md (hunk 2 above)
Comment (S-Mackenzie1678) [inline robot/rookie/c++/README.md]: Not EVERY program.

File: robot/rookie/c++/README.md (hunk 1 above)
Comment (S-Mackenzie1678) [inline robot/rookie/c++/README.md]: The function is `raw_input()`

File: robot/rookie/c++/README.md (hunk 8 above)
Comment (S-Mackenzie1678) [reply robot/rookie/c++/README.md]: It's actually ``` for multiple lines.

File: robot/rookie/c++/README.md [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+Here you will learn the basics of the main language that robot programming uses, C++
+
+###C++ Program Structure
+
+Every C++ program starts with *include statements*. They let you use functions and classes that have been declared in other *header files*. 
+  example:
+    
+    #include <iostream>
+
+(The header `<iostream>` gives you access to input and output helpers, like std::cin and std::out. )
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: `std::cout`, not `std::out`

File: robot/rookie/c++/README.md [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+
+Including `<iostream>` allows you to use `std::cin` and `std::cout`. `std::cout` is the C++ equivalent of the `print()` function in Python. `std::cin` is the C++ equivalent of the `raw_input()` function. 
+
+**More Misc. Initialization Syntax** 
+
+    TypeName var_name = to_copy;
+    TypeName var_name {param1, param2};
+    auto var_name = TypeName{param1, param2};
+
+(this is pseudocode)
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: explain what this means or give an example (and the equivalent in python)

File: robot/rookie/c++/README.md [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+Python: 
+
+     class Person:
+       // Initialize firstName and lastName
+       def __init__(self, first, last):
+         self.firstName = first
+         self.lastName = last
+
+      def greet(self,name):
+         return "Hi, “,name,”! I'm ”,self.firstName, self.lastName,”!"
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: add spaces after commas

Comment (nathansolomon1678) [reply robot/rookie/c++/README.md]: All python code should use 4 spaces for indentation

File: robot/rookie/c++/README.md (hunk 5 above)
Comment (nathansolomon1678) [reply robot/rookie/c++/README.md]: be sure to mention that somewhere

File: robot/rookie/c++/README.md [hunk 16]
```diff
@@ -0,0 +1,154 @@
...
+**Dynamic-size Arrays**
+
+A dynamic size array is an array that can be added to or taken away from. They are used to store values. 
+C++:
+
+    auto nums = std::vector<int> {1,2,3,5,7,11};
+    std::vector<std::string> strings = {"hello", "how are you", "goodbye"};
+
+    // Iteration
+for (int n = 0; n < 8; n++){
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: indentation

Comment (nathansolomon1678) [reply robot/rookie/c++/README.md]: sorry for being picky, but there should be a space before the `{`

File: robot/rookie/c++/README.md [hunk 17]
```diff
@@ -0,0 +1,154 @@
...
+
+      std::cout << n << std::endl;
+    }
+
+    // Check size
+    std::cout << nums.size() << std::endl;
+
+    nums.push_back(7); // Add value
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: 2 spaces between the end of the line and `//`

File: robot/rookie/c++/README.md [hunk 18]
```diff
@@ -0,0 +1,154 @@
...
+
+Python:
+
+    nums = [1,2,3,5,7,11]
```
Comment (nathansolomon1678) [inline robot/rookie/c++/README.md]: pretty much all commas should have a space afterwards

File: robot/rookie/c++/README.md (hunk 5 above)
Comment (ElspethBair) [reply robot/rookie/c++/README.md]: I mention it in line 119.

**Pull request merged**